adapters/emtv/emtv.go

```diff
@@ -0,0 +1,158 @@
+package emtv
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/util/httputil"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type reqBodyExt struct {
+	EmtvBidderExt reqBodyExtBidder `json:"bidder"`
+}
+
+type reqBodyExtBidder struct {
+	Type        string `json:"type"`
+	PlacementID string `json:"placementId,omitempty"`
+	EndpointID  string `json:"endpointId,omitempty"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var err error
+	var adapterRequests []*adapters.RequestData
+
+	reqCopy := *request
+	for _, imp := range request.Imp {
+		reqCopy.Imp = []openrtb2.Imp{imp}
+
+		var bidderExt adapters.ExtImpBidder
+		var emtvExt openrtb_ext.ImpExtEmtv
+
+		if err = json.Unmarshal(reqCopy.Imp[0].Ext, &bidderExt); err != nil {
+			return nil, []error{err}
+		}
+		if err = json.Unmarshal(bidderExt.Bidder, &emtvExt); err != nil {
+			return nil, []error{err}
+		}
+
+		impExt := reqBodyExt{EmtvBidderExt: reqBodyExtBidder{}}
+
+		if emtvExt.PlacementID != "" {
+			impExt.EmtvBidderExt.PlacementID = emtvExt.PlacementID
+			impExt.EmtvBidderExt.Type = "publisher"
+		} else if emtvExt.EndpointID != "" {
+			impExt.EmtvBidderExt.EndpointID = emtvExt.EndpointID
+			impExt.EmtvBidderExt.Type = "network"
+		} else {
+			continue
+		}
+
+		finalyImpExt, err := json.Marshal(impExt)
+		if err != nil {
+			return nil, []error{err}
+		}
+
+		reqCopy.Imp[0].Ext = finalyImpExt
+
+		adapterReq, err := a.makeRequest(&reqCopy)
+		if err != nil {
+			return nil, []error{err}
+		}
```

Comment:
This will abort entire request if error is seen for any one iteration of `request.Imp`. Is this expected behaviour? or do we want to discard current error throwing imp and continue processing next imp. 

We could use `continue` and collect errors  if expected behaviour is to continue processing next imp. 

```
var errs [] error
var adapterRequests []*adapters.RequestData

reqCopy := *request
for _, imp := range request.Imp {
	reqCopy.Imp = []openrtb2.Imp{imp}

	var bidderExt adapters.ExtImpBidder
	var emtvExt openrtb_ext.ImpExtEmtv

	if err = json.Unmarshal(reqCopy.Imp[0].Ext, &bidderExt); err != nil {
		errs = append(errs, err)
		continue
	}
	if err = json.Unmarshal(bidderExt.Bidder, &emtvExt); err != nil {
		errs = append(errs, err)
		continue
	}

	impExt := reqBodyExt{EmtvBidderExt: reqBodyExtBidder{}}

	if emtvExt.PlacementID != "" {
		impExt.EmtvBidderExt.PlacementID = emtvExt.PlacementID
		impExt.EmtvBidderExt.Type = "publisher"
	} else if emtvExt.EndpointID != "" {
		impExt.EmtvBidderExt.EndpointID = emtvExt.EndpointID
		impExt.EmtvBidderExt.Type = "network"
	} else {
		errs = append(errs, errors.New("placement_id or endpoint_id not found"))
		continue
	}

	finalyImpExt, err := json.Marshal(impExt)
	if err != nil {
		errs = append(errs, err)
		continue
	}

	reqCopy.Imp[0].Ext = finalyImpExt

	adapterReq, err := a.makeRequest(&reqCopy)
	if err != nil {
		errs = append(errs, err)
		continue
	}

	if adapterReq != nil {
		adapterRequests = append(adapterRequests, adapterReq)
	}
}

if len(adapterRequests) == 0 {
	return nil, errs
}

return adapterRequests, errs
```